TestBallastMemory randomly fails still, even with the old fix

**Describe the bug**
TestMemoryBallast fails.  The output of the test failure is meaningless as well.  It does not always fail, but I just saw one fail in a pull request because the functionality we wrote to "give it some leeway" said "I expected nothing higher than 88, the value was 89" and it failed the whole PR.

**Steps to reproduce**
If possible, provide a recipe for reproducing the error.
Just make a PR, run the testbed, it will eventually fail.
**What did you expect to see?**
A clear and concise description of what you expected to see.
I expect to see information on what we are really testing here, and what do the values mean.  Yes 89 is higher than 88, I sincerely doubt that thats a problem of memory usage.

We should probably look and test and see if Memory values are a deviation off after multiple runs of a piece of a code, right now it just seems like it give its it an extra 10% leeway.  That might hide real issues as well.  I'd like to discuss the test here a little bit, what it does, why its there and find a real solution to this issue.

If any test fails, it should be able to convey to me a path to resolution.  If a contributor saw the memory ballast was impacted by their changes and this test failed there is no detail on what one might do to go about remediating this. 
**What did you see instead?**
A clear and concise description of what you saw instead.
That some extra memory was being used for some reason, very little, but more than we expected
**What version did you use?**
Version: (e.g., `v0.4.0`, `1eb551b`, etc)
Any build
**What config did you use?**
Config: (e.g. the yaml config file)
Any
**Environment**
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")
Any
**Additional context**
Old issue here 
https://github.com/open-telemetry/opentelemetry-collector/issues/3233

The log from one of my runs 

```
=== RUN   TestBallastMemory
2021/12/06 14:17:47 Starting Agent (/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/bin/otelcontribcol_linux_amd64)
2021/12/06 14:17:47 Writing Agent log to /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/tests/results/TestBallastMemory/agent.log
2021/12/06 14:17:47 Agent running, pid=1858
2021/12/06 14:17:47 Stopped generator. Sent:         0 items
2021/12/06 14:17:47 Gracefully terminating Agent pid=1858, sending SIGTEM...
2021/12/06 14:17:47 Stopping process monitor.
2021/12/06 14:17:47 Agent process stopped, exit code=0
2021/12/06 14:17:47 Starting Agent (/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/bin/otelcontribcol_linux_amd64)
2021/12/06 14:17:47 Writing Agent log to /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/tests/results/TestBallastMemory/agent.log
2021/12/06 14:17:47 Agent running, pid=1865
    assertion_compare.go:342: 
        	Error Trace:	e2e_test.go:122
        	Error:      	"89" is not less than or equal to "88"
        	Test:       	TestBallastMemory
        	Messages:   	[]
2021/12/06 14:17:47 Stopped generator. Sent:         0 items
2021/12/06 14:17:47 Gracefully terminating Agent pid=1865, sending SIGTEM...
2021/12/06 14:17:47 Stopping process monitor.
2021/12/06 14:17:47 Agent process stopped, exit code=0
2021/12/06 14:17:47 Starting Agent (/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/bin/otelcontribcol_linux_amd64)
2021/12/06 14:17:47 Writing Agent log to /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/testbed/tests/results/TestBallastMemory/agent.log
2021/12/06 14:17:47 Agent running, pid=1872
2021/12/06 14:17:47 Stopped generator. Sent:         0 items
2021/12/06 14:17:47 Gracefully terminating Agent pid=1872, sending SIGTEM...
2021/12/06 14:17:47 Stopping process monitor.
2021/12/06 14:17:47 Agent process stopped, exit code=0
--- FAIL: TestBallastMemory (0.76s)
```

--- a/testbed/tests/e2e_test.go
+++ b/testbed/tests/e2e_test.go
@@ -74,7 +74,7 @@ func TestBallastMemory(t *testing.T) {
 		maxRSS      uint32
 	}{
 		{100, 70},
-		{500, 80},
+		{500, 90},
 		{1000, 110},
 	}
 
@@ -84,43 +84,45 @@ func TestBallastMemory(t *testing.T) {
 	options := testbed.LoadOptions{DataItemsPerSecond: 10_000, ItemsPerBatch: 10}
 	dataProvider := testbed.NewPerfTestDataProvider(options)
 	for _, test := range tests {
-		sender := testbed.NewOTLPTraceDataSender(testbed.DefaultHost, testbed.GetAvailablePort(t))
-		receiver := testbed.NewOTLPDataReceiver(testbed.GetAvailablePort(t))
-		ballastCfg := createConfigYaml(
-			t, sender, receiver, resultDir, nil,
-			map[string]string{"memory_ballast": fmt.Sprintf(ballastConfig, test.ballastSize)})
-		cp := testbed.NewChildProcessCollector()
-		cleanup, err := cp.PrepareConfig(ballastCfg)
-		require.NoError(t, err)
-		tc := testbed.NewTestCase(
-			t,
-			dataProvider,
-			sender,
-			receiver,
-			cp,
-			&testbed.PerfTestValidator{},
-			performanceResultsSummary,
-			testbed.WithSkipResults(),
-			testbed.WithResourceLimits(testbed.ResourceSpec{ExpectedMaxRAM: test.maxRSS}),
-		)
-		tc.StartAgent()
+		t.Run(fmt.Sprintf("ballast-size-%d", test.ballastSize), func(t *testing.T) {
+			sender := testbed.NewOTLPTraceDataSender(testbed.DefaultHost, testbed.GetAvailablePort(t))
+			receiver := testbed.NewOTLPDataReceiver(testbed.GetAvailablePort(t))
+			ballastCfg := createConfigYaml(
+				t, sender, receiver, resultDir, nil,
+				map[string]string{"memory_ballast": fmt.Sprintf(ballastConfig, test.ballastSize)})
+			cp := testbed.NewChildProcessCollector()
+			cleanup, err := cp.PrepareConfig(ballastCfg)
+			require.NoError(t, err)
+			tc := testbed.NewTestCase(
+				t,
+				dataProvider,
+				sender,
+				receiver,
+				cp,
+				&testbed.PerfTestValidator{},
+				performanceResultsSummary,
+				testbed.WithSkipResults(),
+				testbed.WithResourceLimits(testbed.ResourceSpec{ExpectedMaxRAM: test.maxRSS}),
+			)
+			tc.StartAgent()
 
-		var rss, vms uint32
-		// It is possible that the process is not ready or the ballast code path
-		// is not hit immediately so we give the process up to a couple of seconds
-		// to fire up and setup ballast. 2 seconds is a long time for this case but
-		// it is short enough to not be annoying if the test fails repeatedly
-		tc.WaitForN(func() bool {
-			rss, vms, _ = tc.AgentMemoryInfo()
-			return vms > test.ballastSize
-		}, time.Second*2, fmt.Sprintf("VMS must be greater than %d", test.ballastSize))
+			var rss, vms uint32
+			// It is possible that the process is not ready or the ballast code path
+			// is not hit immediately so we give the process up to a couple of seconds
+			// to fire up and setup ballast. 2 seconds is a long time for this case but
+			// it is short enough to not be annoying if the test fails repeatedly
+			tc.WaitForN(func() bool {
+				rss, vms, _ = tc.AgentMemoryInfo()
+				return vms > test.ballastSize
+			}, time.Second*2, fmt.Sprintf("VMS must be greater than %d", test.ballastSize))
 
-		// https://github.com/open-telemetry/opentelemetry-collector/issues/3233
-		// given that the maxRSS isn't an absolute maximum and that the actual maximum might be a bit off,
-		// we give some room here instead of failing when the memory usage isn't that much higher than the max
-		lenientMax := 1.1 * float32(test.maxRSS)
-		assert.LessOrEqual(t, float32(rss), lenientMax)
-		cleanup()
-		tc.Stop()
+			// https://github.com/open-telemetry/opentelemetry-collector/issues/3233
+			// given that the maxRSS isn't an absolute maximum and that the actual maximum might be a bit off,
+			// we give some room here instead of failing when the memory usage isn't that much higher than the max
+			lenientMax := 1.1 * float32(test.maxRSS)
+			assert.LessOrEqual(t, float32(rss), lenientMax, fmt.Sprintf("The RSS memory usage (%d) is >10%% higher than the limit (%d).", rss, test.maxRSS))
+			cleanup()
+			tc.Stop()
+		})
 	}
 }

